### ISSUE TITLE
As a user I want to snap vertices and subfiles to the current grid.

### ISSUE THREAD

**Issue by nilsschmidt1337** [nilsschmidt1337/ldparteditor#987]
1. Snap vertices to grid (Snap to Grid)
2. ~Snap vertices to "move snapping" (Snap to Move Step)~ (don't implement it yet)

```
var gridSize = 2f;
var value = 2.1f;
var result = value - (value % gridSize);
System.out.println("Snapped (2.0f)" + result);

// Precise
new BigDecimal(value).subtract(new BigDecimal(value).remainder(gridSizePrecise, Threshold.MC));
```

**Comment by nilsschmidt1337** [comment]
Closed with commit [4455423](https://github.com/nilsschmidt1337/ldparteditor/commit/4455423932728c73bc2934469ac2085f7156a582).

**Comment by nilsschmidt1337** [comment]
It should also snap subfiles.

**Comment by nilsschmidt1337** [comment]
Closed with commit [6aa330a](https://github.com/nilsschmidt1337/ldparteditor/commit/6aa330a486e253cd1063a10a5c79b5d4e93e2dd8).
